### PR TITLE
fix(dispatcher): unbounded overload batch_size

### PR DESCRIPTION
**Pull request by aarnphm**
## What does this PR address?

In the training logics, we might override the batch_size variable here, causing `UnboundedLocalError: local variable 'batch_size' referenced before assignment`. This PR aims to fix that
by copying the batch_size variable to a new variable.

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml/_internal/marshal/dispatcher.py [hunk 1]
```diff
@@ -227,18 +232,18 @@ async def train_optimizer(
 
                 if self.max_batch_size == -1:  # batching is disabled
                     n_call_out = 1
-                    batch_size = self._queue[0].data.sample.batch_size
+                    training_batch_size = self._queue[0].data.sample.batch_size
```
Comment (sauyon) [inline src/bentoml/_internal/marshal/dispatcher.py:235]: These should not be changed?

File: src/bentoml/_internal/marshal/dispatcher.py [hunk 2]
```diff
@@ -190,13 +190,15 @@ async def train_optimizer(
         num_reqs_to_train: int,
         batch_size: int,
```
Comment (sauyon) [inline src/bentoml/_internal/marshal/dispatcher.py:191]: You need to correct the argument too!

Comment (aarnphm) [reply src/bentoml/_internal/marshal/dispatcher.py:191]: But we are assigning this batch_size below right? where

Comment (sauyon) [reply src/bentoml/_internal/marshal/dispatcher.py:191]: Why are you doing that? Just rename the argument, this is just less clear what's happening imo.

**Pull request merged**